### ISSUE TITLE
Clarify differences between overlay and dmsquash for live images

### ISSUE THREAD

**Issue by Vogtinator** [OSInside/kiwi#1156]
Currently it's not obvious which features are actually supported and in which ways.

I just built images with dmsquash instead of overlay and they seem to work fine, but even though persistence is mentioned on the kernel cmdline, it does not work OOTB as it did with overlay.

**Comment by schaefi** [comment]
The iso structure as kiwi builds it is compatible with dmsquash and overlay. We did this on purpose to allow the use of both modules.

* dmsquash is the dracut upstream module. It uses the device mapper as preferred method
* overlay is the kiwi provided module. It uses only overlayfs as provided by the kernel

I had a look into the dmsquash module upstream to see if I could easily include support for overlayfs. But the risk to break anything there was very high and I don't wanted to introduce patch number 4711 in the dracut package of suse ;)

This is the reason why two modules exists. For the overlayfs based code I took care that all supported features e.g persistent writing works, I can't tell that for the dmsquash version though as it is maintained by other people

Hope that explains the current situation. I agree it might be confusing to see two different modules for the same purpose but both are based on different overlay technologies. From a future perspective I think overlayfs will be the successor for all the device mapper setup since it's easier to handle and more clear in coding. If this will settle on the upstream module it would be imho a good move. I just didn't had the time to get in contact with dracut upstream to make a suggestion
